docker/docker_client.go

```diff
@@ -535,7 +535,7 @@ func (c *dockerClient) getBearerToken(ctx context.Context, challenge challenge,
 	case http.StatusUnauthorized:
 		err := client.HandleErrorResponse(res)
 		logrus.Debugf("Server response when trying to obtain an access token: \n%q", err.Error())
-		return nil, ErrUnauthorizedForCredentials
+		return nil, errors.Wrapf(ErrUnauthorizedForCredentials, err.Error())
```

Comment:
This is not really how `Wrapf` is supposed to be used: `Wrapf` only adds a string.

I suppose we want _both_:
- A well-typed `ErrUnauthorizedForCredentials` that can be recognized by callers
- Preserving the original typed error value returned by `client.HandleErrorResponse`, not just its string representation.

The way `types.ManifestTypeRejectedError` is defined seems like the simplest way to do it.